Core/include/Acts/TrackFinding/MeasurementSelector.ipp

```diff
@@ -28,24 +28,17 @@ MeasurementSelector::select(
 
   // Get geoID of this surface
   auto geoID = candidates.front().referenceSurface().geometryId();
+  // Get the eta of the first track state
+  const auto eta =
+      std::atanh(std::cos(candidates.front().predicted()[eBoundTheta]));
```

Comment:
The previous code only calculated `eta` if there were multiple eta bins, and so far, I think we have only used cuts that are not eta-dependent. This is a potential pessimisation.
I just realised that in Athena, when we harmonised all the eta binning, we picked up multiple eta bins also for the `MeasurementSelector`, so that might need reoptimising. That won't help if here we calculate eta even if it isn't needed.

Reply:
I can special case this again